### ISSUE TITLE
Installing M2Crypto 0.30 on a Windows minion stops the master's public key being authenticated

### ISSUE THREAD

**Issue by mts-avco** [saltstack/salt#48994]
### Description of Issue/Question
I've compiled a wheel package for M2Crypto 0.30.1 for py3.5 (win64) but when installed it causes the minion to stop accepting the master's public key. 

```
C:\salt>salt-call test.ping
[CRITICAL] The Salt Master server's public key did not authenticate!
The master may need to be updated if it is a version of Salt lower than 2018.3.2, or
If you are confident that you are connecting to a valid Salt Master, then remove the master public key and restart the Salt Minion.
The master public key can be found at:
c:\salt\conf\pki\minion\minion_master.pub
```
Removing the key makes no difference, although uninstalling M2Crypto resolves this immediately. 

### Setup
Windows minion connected successfully to master, both at 2018.3.2. Stop the minion service, then create and install M2Crypto 0.30.1 package using OpenSSL 1.1.0h as below.

### Steps to Reproduce Issue
To create the package I used the following steps (based on https://gitlab.com/m2crypto/m2crypto/issues/213)

- install swigwin (version 3.0.12) and add swig.exe to the current path
- install Win32/64OpenSSL (version 1.1.0h Win64 full)
- "cd /d C:\salt\bin\Scripts"
- "pip download M2Crypto" and extract `M2Crypto-0.30.1.tar.gz` into an empty directory
- cd to that directory
- "C:\salt\bin\python setup.py build --openssl={winopenssl-dir}"
- copy the two OpenSSL binaries into the M2Crypto folder
- "C:\salt\bin\python setup.py bdist_wheel"
- "cd /d C:\salt\bin\Scripts"
- "pip install {dist/m2c-wheel-file}"

### Versions Report
```
Salt Version:
           Salt: 2018.3.2

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.30.1
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: 2.0.3
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist:
         locale: cp1252
        machine: AMD64
        release: 10
         system: Windows
        version: 10 10.0.17134 SP0 Multiprocessor Free
```

**Comment by mts-avco** [comment]
I've looked at #46868 and I don't believe this is the same problem. None of the public keys start with a line containing `RSA` and none of the suggestions in that issue seem to apply here, but I'm happy to try out any testing.

**Comment by mts-avco** [comment]
Having just been pointed to it, this appears to be resolved by PR #48358
